finagle-core/src/main/scala/com/twitter/finagle/dispatch/ServerDispatcher.scala

```diff
@@ -103,8 +114,30 @@ class SerialServerDispatcher[Req, Rep](
     service.close()
   }
 
-  protected def dispatch(req: Req, eos: Promise[Unit]) =
+  override def dispatch(req: Req, eos: Promise[Unit]) =
     service(req) ensure eos.setDone()
 
-  protected def handle(rep: Rep) = trans.write(rep)
+  override def handle(rep: Rep) = trans.write(rep)
+}
+
+/**
+  * A wrapper type around a 
+  * [[com.twitter.finagle.dispatcher.ServerDispather ServerDispatcher]]
```

Comment:
OK, let me be more precise.  There are at least three models of request / response interactions that we're considering in finagle right now.  We support two of them, byte streaming (over http) and object RPC today, and we're considering support for a third, which would be object streaming.

AnnotationServerDispatcher is appropriate for object RPC (we have everything we need as soon as we write the first byte to the wire), but it's probably not appropriate for either object streaming or byte streaming.